### PR TITLE
Site workflow module

### PR DESCRIPTION
**Pull request by sonalideepthi777**
## Description
[CSCwn25085] --> iac2.0: sites: Backward compatibility broken, 2.3.5.3 sites config not working.
[CSCwn25842]. --> [SITE]: Error handling for "rf_model" not existing

## Type of Change
- [x] Bug fix
- [ ] New feature
- [ ] Breaking change
- [x] Documentation update

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] All the sanity checks have been completed and the sanity test cases have been executed

## Ansible Best Practices
- [ ] Tasks are idempotent (can be run multiple times without changing state)
- [ ] Variables and secrets are handled securely (e.g., using `ansible-vault` or environment variables)
- [x] Playbooks are modular and reusable
- [ ] Handlers are used for actions that need to run on change

## Documentation
- [x] All options and parameters are documented clearly.
- [x] Examples are provided and tested.
- [x] Notes and limitations are clearly stated.

### CODE REVIEW

**Review by madhansansel — APPROVED** [review]

File: plugins/modules/site_workflow_manager.py [hunk 1]
```diff
@@ -125,9 +125,10 @@
                 description: |
                     Floor number within the building site (e.g., 5). This value can only be specified during the creation of the
                     floor and cannot be modified afterward.
```
Comment (madhansansel) [inline plugins/modules/site_workflow_manager.py:127]: It is required from version 2.3.7.6 onwards.

File: plugins/modules/site_workflow_manager.py [hunk 2]
```diff
@@ -125,9 +125,10 @@
...
+                    This field is mandatory in 2.3.7.6 version.
                 type: int
               units_of_measure:
-                description: The unit of measurement for floor dimensions, typically 'feet' or 'meters'.
+                description: The unit of measurement for floor dimensions, typically 'feet' or 'meters'. This field is introduced in 2.3.7.6 version.
```
Comment (madhansansel) [inline plugins/modules/site_workflow_manager.py:131]: Specifies the unit of measurement for floor dimensions, typically 'feet' or 'meters'. This field is introduced from version 2.3.7.6 onwards.

File: plugins/modules/site_workflow_manager.py [hunk 3]
```diff
@@ -1287,15 +1222,20 @@ def validate_site_input_data(self, config):
...
-                    errormsg.append("Either Latitude or longitude is missing in the given playbook")
+                if self.compare_dnac_versions(self.get_ccc_version(), "2.3.7.6") >= 0:
+                    if not (latitude and longitude or address):
+                        errormsg.append("Either latitude/longitude or address is required.")
+                        self.log("Missing required latitude/longitude or address for building.", "ERROR")
+                    elif (latitude and not longitude) or (not latitude and longitude):
+                        errormsg.append("Either Latitude or longitude is missing in the given playbook")
+                else:
+                    if not (latitude and longitude):
+                        errormsg.append("latitude and longitude is required.")
```
Comment (madhansansel) [inline plugins/modules/site_workflow_manager.py:1233]: errormsg.append("Latitude and longitude are required.")

File: plugins/modules/site_workflow_manager.py [hunk 4]
```diff
@@ -1304,18 +1244,19 @@ def validate_site_input_data(self, config):
...
-                if floor_number:
-                    self.log("Validating 'floor_number': " + str(floor_number), "DEBUG")
-                    if not (isinstance(floor_number, int) and -200 <= floor_number <= 200):
-                        errormsg.append("Please enter a valid floor number (-200 to 200)")
-                        self.log("Missing 'floor_number' in floor entry.", "ERROR")
-                else:
-                    errormsg.append("Floor number should not be None or empty")
+                if self.compare_dnac_versions(self.get_ccc_version(), "2.3.7.6") >= 0:
+                    floor_number = site.get(site_type, {}).get("floor_number")
+                    if floor_number:
```
Comment (madhansansel) [inline plugins/modules/site_workflow_manager.py:1249]: if floor_number is not None:

floor_number could be 0 also?

File: plugins/modules/site_workflow_manager.py [hunk 5]
```diff
@@ -1304,18 +1244,19 @@ def validate_site_input_data(self, config):
...
+                        self.log("Validating 'floor_number': " + str(floor_number), "DEBUG")
+                        if not (isinstance(floor_number, int) and -200 <= floor_number <= 200):
+                            errormsg.append("Please enter a valid floor number (-200 to 200)")
+                            self.log("Missing 'floor_number' in floor entry.", "ERROR")
+                    else:
+                        errormsg.append("Floor number should not be None or empty")
```
Comment (madhansansel) [inline plugins/modules/site_workflow_manager.py:1255]: errormsg.append("'floor_number' should not be None or empty.")
        self.log("Missing 'floor_number' in floor entry.", "ERROR")

File: plugins/modules/site_workflow_manager.py [hunk 6]
```diff
@@ -1304,18 +1244,19 @@ def validate_site_input_data(self, config):
...
-                        self.log("Missing 'floor_number' in floor entry.", "ERROR")
-                else:
-                    errormsg.append("Floor number should not be None or empty")
+                if self.compare_dnac_versions(self.get_ccc_version(), "2.3.7.6") >= 0:
+                    floor_number = site.get(site_type, {}).get("floor_number")
+                    if floor_number:
+                        self.log("Validating 'floor_number': " + str(floor_number), "DEBUG")
+                        if not (isinstance(floor_number, int) and -200 <= floor_number <= 200):
+                            errormsg.append("Please enter a valid floor number (-200 to 200)")
+                            self.log("Missing 'floor_number' in floor entry.", "ERROR")
```
Comment (madhansansel) [inline plugins/modules/site_workflow_manager.py:1253]: self.log("'floor_number' is out of the valid range (-200 to 200).", "ERROR")

File: plugins/modules/site_workflow_manager.py [hunk 7]
```diff
@@ -1353,14 +1294,15 @@ def validate_site_input_data(self, config):
...
-                                units_of_measure))
-                else:
-                    errormsg.append("units_of_measure should not be None or empty")
+                if self.compare_dnac_versions(self.get_ccc_version(), "2.3.7.6") >= 0:
+                    units_of_measure = site.get(site_type, {}).get("units_of_measure")
+                    if units_of_measure:
+                        if units_of_measure not in ("feet", "meters"):
+                            errormsg.append(
+                                "units_of_measure: Invalid value '{0}' for units_of_measure in playbook. Must be one of 'feet' or 'meters'.".format(
+                                    units_of_measure))
```
Comment (madhansansel) [inline plugins/modules/site_workflow_manager.py:1303]: self.log("Invalid 'units_of_measure': {0}. Expected 'feet' or 'meters'.".format(units_of_measure), "ERROR")

File: plugins/modules/site_workflow_manager.py [hunk 8]
```diff
@@ -1353,14 +1294,15 @@ def validate_site_input_data(self, config):
...
+                    else:
+                        errormsg.append("units_of_measure should not be None or empty")
```
Comment (madhansansel) [inline plugins/modules/site_workflow_manager.py:1305]: self.log("Missing 'units_of_measure' in floor entry.", "ERROR")

File: plugins/modules/site_workflow_manager.py (hunk 1 above)
Comment (sonalideepthi777) [reply plugins/modules/site_workflow_manager.py:127]: Modified

File: plugins/modules/site_workflow_manager.py (hunk 2 above)
Comment (sonalideepthi777) [reply plugins/modules/site_workflow_manager.py:131]: Modified

File: plugins/modules/site_workflow_manager.py (hunk 3 above)
Comment (sonalideepthi777) [reply plugins/modules/site_workflow_manager.py:1233]: Modified

File: plugins/modules/site_workflow_manager.py (hunk 5 above)
Comment (sonalideepthi777) [reply plugins/modules/site_workflow_manager.py:1255]: Added

File: plugins/modules/site_workflow_manager.py (hunk 6 above)
Comment (sonalideepthi777) [reply plugins/modules/site_workflow_manager.py:1253]: Modified

File: plugins/modules/site_workflow_manager.py (hunk 7 above)
Comment (sonalideepthi777) [reply plugins/modules/site_workflow_manager.py:1303]: Added

File: plugins/modules/site_workflow_manager.py (hunk 8 above)
Comment (sonalideepthi777) [reply plugins/modules/site_workflow_manager.py:1305]: Added

**Review by sonalideepthi777 — COMMENTED** [review]
Comments fixed

**Review by madhansansel — APPROVED** [review]

**Pull request merged**